operators/pkg/forge/ingresses.go

```diff
@@ -126,17 +134,48 @@ func HostName(baseHostName string, mode clv1alpha2.EnvironmentMode) string {
 	return baseHostName
 }
 
-// IngressInstancePath returns the path of the ingress targeting the environment.
-func IngressInstancePath(instance *clv1alpha2.Instance) string {
-	return strings.TrimRight(fmt.Sprintf("%v/%v", IngressInstancePrefix, instance.UID), "/")
+// IngressMyDrivePath returns the path of the ingress targeting the environment "MyDrive".
+func IngressMyDrivePath(instance *clv1alpha2.Instance) string {
+	return strings.TrimRight(fmt.Sprintf("%v/%v/%v", IngressInstancePrefix, instance.UID, IngressMyDrivePathSuffix), "/")
 }
 
-// IngressVNCGUIPath returns the path of the ingress targeting the environment GUI vnc.
-func IngressVNCGUIPath(instance *clv1alpha2.Instance) string {
-	return strings.TrimRight(fmt.Sprintf("%v/%v/%v", IngressInstancePrefix, instance.UID, IngressVNCGUIPathSuffix), "/")
+// IngressGUIPath returns the path of the ingress targeting the environment GUI vnc or Standalone.
+func IngressGUIPath(instance *clv1alpha2.Instance, environment *clv1alpha2.Environment) string {
+	switch environment.EnvironmentType {
+	case clv1alpha2.ClassStandalone:
+		if environment.RewriteURL {
+			return strings.TrimRight(fmt.Sprintf("%v/%v/%v", IngressInstancePrefix, instance.UID, IngressStandaloneSuffix+"(/|$)(.*)"), "/")
+		}
+		return strings.TrimRight(fmt.Sprintf("%v/%v/%v", IngressInstancePrefix, instance.UID, IngressStandaloneSuffix), "/")
+	case clv1alpha2.ClassContainer, clv1alpha2.ClassCloudVM, clv1alpha2.ClassVM:
+		return strings.TrimRight(fmt.Sprintf("%v/%v/%v", IngressInstancePrefix, instance.UID, IngressVNCGUIPathSuffix), "/")
+	}
+	return ""
 }
 
-// IngressMyDrivePath returns the path of the ingress targeting the environment "MyDrive".
-func IngressMyDrivePath(instance *clv1alpha2.Instance) string {
-	return strings.TrimRight(fmt.Sprintf("%v/%v/%v", IngressInstancePrefix, instance.UID, IngressMyDrivePathSuffix), "/")
+// IngressGUICleanPath returns the path of the ingress targeting the environment GUI vnc or Standalone, without the url-rewrite's regex.
+func IngressGUICleanPath(instance *clv1alpha2.Instance) string {
+	return strings.TrimRight(fmt.Sprintf("%v/%v/%v", IngressInstancePrefix, instance.UID, IngressStandaloneSuffix), "/")
+}
+
+// IngressGuiStatusURL returns the path of the ingress targeting the environment.
+func IngressGuiStatusURL(host string, environment *clv1alpha2.Environment, instance *clv1alpha2.Instance) string {
+	switch environment.EnvironmentType {
+	case clv1alpha2.ClassStandalone:
+		return "https://" + host + fmt.Sprintf("%v/%v/%v/", IngressInstancePrefix, instance.UID, IngressStandaloneSuffix)
+	case clv1alpha2.ClassContainer, clv1alpha2.ClassVM, clv1alpha2.ClassCloudVM:
+		return "https://" + host + fmt.Sprintf("%v/%v/", IngressInstancePrefix, instance.UID)
```

Comment:
Would perform the entire concatenation through `fmt.Sprintf`, without mixing it with "adding" strings.

Reply:
Done